### PR TITLE
adds char encoding in html example

### PR DESCRIPTION
**Pull request by djrecipe**
IronPdf new renderer is based on Chrome. If you try open_me_in_chrome_browser.html in chrome you will find it does not render the Chinese properly. Chrome browser checks the first 60,000 or so characters to auto-detect encoding and thus a very long HTML string with special chars at the end (but not beginning) will fail the encoding detection in Chrome. Therefor it is best practice to specify encoding at the start of your html (<meta charset='utf-8'/>) to ensure it works in all browsers.

However, the upcoming IronPdf release (within a week or so) will attempt to detect the absence of explicit character encoding and will automatically prepend <meta charset='utf-8'/> to your html so this issue should go away

### CODE REVIEW

**Comment by AgWillo** [comment]
Thanks, @djrecipe. Inasmuch as the code in this repo reflects what's in our app, the fix (adding the `<meta>` tag in lieu of the BOM) is better done as part of the PDF-generation logic. The method that generates the HTML is also used to provide a browser preview of its result, in which case the page's character encoding has already been established by other means.

**Comment by djrecipe** [comment]
Thanks for bringing this to my/our attention - it's a pretty interesting issue that I am also submitting to the Chromium bug tracker on Bitbucket.

We initialize an internal Chromium browser instance with default encoding of "UTF-8", but this bug seems to occur regardless of browser "default encoding". I performed some tests which simply prepends the meta text to the beginning of the HTML if it was not specified by the user, however, this negatively impacts PDF generation, causing issues in the generated PDF in some cases. We are working on a more intelligent solution which will traverse the entire DOM of the HTML to search for the element encoding specification element and properly insert it into the head section of your HTML if it does not exist already. This is something we (Iron Software) is working on as I can totally understand how this would be frustrating to yourself and others who have long HTML strings.

I'd like to reiterate that the current behavior is the base behavior of the Chromium browser (the Chinese characters do not render properly if only appearing at the end of a very long HTML string). However, we believe that the many advantages of Chromium browser outweigh the disadvantages.

In conclusion, we are implementing a workaround for this bug/limitation of Chromium as we definitely think that your HTML is valid and should render properly without having to specify character encoding. However, we have to be sure that this implementation does not negatively impact any other aspect of rendering as modifying HTML DOM provided by the user can be cause issues. For the time being, the only suggestion I can offer is manually specifying the encoding via HTML at the top of your document (or within the <head> section) or perhaps trying the WebKitRenderer legacy renderer within IronPdf.

**Comment by djrecipe** [comment]
Or, funnily enough... You can specify a unicode character at the beginning of your HTML (maybe the invisible word-joiner??) as it seems to trigger the Chromium auto-detection.

**Comment by AgWillo** [comment]
> You can specify a unicode character at the beginning of your HTML ...

Which brings us full circle to our original workaround: prepending a byte-order mark (`'\ufeff'`), the technique we're using in production right now. :smile:
